app/controllers/wikis_controller.rb

```diff
@@ -0,0 +1,67 @@
+class WikisController < ApplicationController
+  include Pundit
+  before_action :authenticate_user!, except: [:show, :index]
+
+  def new
+    @wiki = Wiki.new
+  end
+
+  def index
+    @wikis = policy_scope(Wiki)
+  end
+
+  def create
+    @wiki = current_user.wikis.build(wiki_params)
+
+    if @wiki.save
+      redirect_to @wiki
+    else
+      flash.now[:alert] = "Unable to save wiki. Please try again later"
+      render :new
+    end
+  end
+
+  def show
+    begin
+      @wiki = Wiki.find(params[:id])
```

Comment:
you do the finding in a lot of controller actions, but only in `show` do you rescue if the record isn't found. I'd recommend pulling this finding code into a `before_action` that is run before the actions that need to find `@wiki` from the params, and also includes the `rescue` statement if a record can't be found